### PR TITLE
Improved StreamOutput::hasColorSupport()

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #13759 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Console/Output/StreamOutput.php [hunk 1]
```diff
@@ -92,12 +92,16 @@ protected function doWrite($message, $newline)
...
+        if (DIRECTORY_SEPARATOR == '\\') {
+            return false !== getenv('ANSICON') || 'ON' === getenv('ConEmuANSI');
+        } elseif (!function_exists('posix_isatty')){
+            return false;
         }
 
-        return function_exists('posix_isatty') && @posix_isatty($this->stream);
-        // @codeCoverageIgnoreEnd
+        try {
+            return posix_isatty($this->stream);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Output/StreamOutput.php]: this function doesn't throw any exception...

Comment (staabm) [inline src/Symfony/Component/Console/Output/StreamOutput.php]: I guess he handles a exception getting thrown by a error-handler by transforming a warning,...

Comment (javiereguiluz) [inline src/Symfony/Component/Console/Output/StreamOutput.php]: Do you recommend to close this pull request and to declare the related issue as "won't fix"? Thanks.

Comment (fabpot) [inline src/Symfony/Component/Console/Output/StreamOutput.php]: Yes, I think @staabm is right, this is probably related to a bad error handler.

**Pull request closed without merging**